### DISCUSSION TITLE
Python Debugging not working with extension in development & release build

### DISCUSSION THREAD

**Discussion by sakthivel-axim** [coder/code-server#5962] [General]
I have followed below steps,

```
$ git clone git@github.com:coder/code-server.git 
$ git checkout  v4.9.1 
$ quilt push –a 

$ yarn 
$ yarn build 
$ VERSION=4.9.1 yarn build:vscode 
$ KEEP_MODULES=1 yarn release 
$ cd release 

$ VSCODE_DEV=1 node . -vvv
``` 

Created **launch.json** as follows,

```
{
      "name": "Python: Current File",
      "type": "python",
      "request": "launch",
      "program": "${file}",
      "console": "integratedTerminal"
}
```

When I try to run, doesn't show any error and didn't run the python program as well.

Browser console logs

```
TRACE CommandService#executeCommand workbench.action.debug.start
log.ts:300 TRACE onWillActivateByEvent:  onDebugResolve
log.ts:300 TRACE onWillActivateByEvent:  onDebugResolve:python
2log.ts:300 TRACE CommandService#executeCommand setContext
log.ts:300 TRACE [DEBOUNCE: CodeLensProvide] for vscode-remote://127.0.0.1:9001/home/test/IDE/workspace/blinkLED.py is 250ms
2log.ts:300 TRACE CommandService#executeCommand setContext
log.ts:300 TRACE [DEBOUNCE: CodeLensProvide] for vscode-remote://127.0.0.1:9001/home/test/IDE/workspace/blinkLED.py is 250ms

```

**Comment by jsjoeio** [comment]
Hmm...I just tested and it worked for me. 

1. create python file
2. create launch.json
3. install python extension
4. run and debug -> Python

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/212376743-6c530f32-0d26-4b1d-960c-686f6952965b.png">

**Reply by sakthivel-axim** [reply]
@jsjoeio 
I'm able to run the program but Debug is not working. Are you able to debug the Python program?

I have used extension `ms-python.python-2022.18.2.vsix`

OS : Ubuntu 18.04

**Reply by lg-nineoneninethreenine** [reply]
@jsjoeio same here. I'm on Pop! 22.04 Jammy Jellyfish. The "Run and Debug" button only runs the file in debug mode, perfect for GUI coding/app dev. But the launch.json configuration, the exact one @sakthivel-axim uses doesn't list up in the debug tab as expected. Maybe this is expected?

**Reply by lg-nineoneninethreenine** [reply]
Old config by @sakthivel-axim:
```json
{
      "name": "Python: Current File",
      "type": "python",
      "request": "launch",
      "program": "${file}",
      "console": "integratedTerminal"
}
```

***Updated*** and ***working*** `launch.json` script code for @sakthivel-axim:
```json
{
    "configurations": [
        {
            "name": "Python: Current File",
            "type": "python",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal",
            "justMyCode": true
        }
    ]
}
```

<details>
<summary>Old image using old config</summary>

![Screenshot from 2023-01-23 20-10-28](https://user-images.githubusercontent.com/76845820/214068104-d85d6238-9e8d-4ec8-8785-77f3bd8df0cf.jpg)
</details>


![Screenshot from 2023-01-23 20-59-24(1)](https://user-images.githubusercontent.com/76845820/214080328-894a368c-f088-454a-ac76-75363e9f3f53.jpg)
Image of the test on code-server - updated with the new config mentioned above.

![Screenshot from 2023-01-23 20-16-04](https://user-images.githubusercontent.com/76845820/214082219-748473eb-0932-40b3-8988-82f405ed7b42.png)

Image of the same test on VS Code.

I guess now this can be solved. @sakthivel-axim try the new code I put above, and try using the GUI configuration to add the `launch.json` code from now if to avoid issues with the code.

**Reply by jsjoeio** [reply]
Nice fix @im-coder-lg!! Thank you for helping solve this!

(P.S. haven't said it enough, but appreciate all your help these days!)

**Reply by lg-nineoneninethreenine** [reply]
No worries! Plus this wasn't a fix, I just replaced the JSON code of `launch.json` with the built-in configuration of VS Code so as to make sure it actually works.

**Reply by sakthivel-axim** [reply]
Thanks @im-coder-lg  for your support. I have updated `launch.json`, still debug not working and no errors.

**Reply by lg-nineoneninethreenine** [reply]
> Thanks @im-coder-lg for your support. I have updated `launch.json`, still debug not working and no errors.

No worries. But what do you mean by "debug not working"? I think it's fixed.

**Reply by sakthivel-axim** [reply]
I mean even with updated `launch.json`, program is not running. Actually without `launch.json`, I can run the program and got the output. But, with `launch.json` it's not running in development/release build.

**Reply by lg-nineoneninethreenine** [reply]
Hmm, something beyond me. Honestly, that's unheard of!

**Comment by sakthivel-axim** [comment]
@im-coder-lg , I'm able to debug python in Ubuntu20.04 with same steps. Still no clue on why it is not working on Ubuntu18.04

**Reply by lg-nineoneninethreenine** [reply]
Hmm... Bionic beaver doesn't have the support for that. I mean, 2018 software, right? It makes sense why Focal and Jammy support it.
